server-spi/src/main/java/org/keycloak/models/RoleContainerModel.java

```diff
@@ -47,12 +49,27 @@
 
     boolean removeRole(RoleModel role);
 
-    Set<RoleModel> getRoles();
-    
-    Set<RoleModel> getRoles(Integer firstResult, Integer maxResults);
+    @Deprecated
+    default Set<RoleModel> getRoles() {
```

Comment:
This method is now deprecated but still used across the codebase.

Reply:
This was intentionally left for a separate commit together with the rest of methods like `getClients()`. The PR would be huge if it was done at once. I've the second commit WIP.

Reply:
so maybe it'd be good to add `//TODO`?

Reply:
TODO added to multiple places